### PR TITLE
Technical Spike: Test Visibility in Circle CI

### PR DESCRIPTION
**Pull request by RichardSlater**
#### 📲 What

Add support to the pipeline for converting dart test results Ant/JUnit test-results XML 

#### 🤔 Why
		
Unit tests are not currently visible in the Circle CI UI
		
#### 🛠 How
		
Utilising a 3rd party library, [tojunit](https://github.com/TOPdesk/dart-junitreport), to take the machine-readable output from `flutter test --machine` to generate test-results in dart JSON format and convert to JUnit (i.e. Ant) format.

Important: due to a bug in `flutter test` it's essential that `flutter packages get` is executed before `flutter test` as otherwise `Running "flutter packages get" in project...                        3.4s` will be prepended to the dart JSON.

#### 🕵️‍♂️ How to test

Circle CI executation will contain the following output under the "Test Summary" tab:

> Your job ran 4 tests in flutter with 0 failures

See [Circle CI #209](https://circleci.com/gh/farmsmart/farmsmart-flutter/209)

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -10,6 +10,25 @@ jobs:
...
+          command: mkdir --parents /tmp/test-results/flutter
+
+      - run:
+          name: Get Flutter packages to avoid stdout in JSON
+          command: flutter packages get
+
       - run:
           name: Run Flutter tests.
-          command: flutter test
+          command: flutter test --machine > /tmp/test-results/flutter/TEST-report.json
```
Comment (raubreysmith) [inline .circleci/config.yml:39]: Is there some documentation for the --machine param? My Google foo is failing, I'm wondering what other params might be available outside of -h docs

**Review by maribel4amido — DISMISSED** [review]

**Review by raubreysmith — DISMISSED** [review]
Currently the error output includes \n characters - this is output by dart in the json format. I've raised an issue with the conversion library to see if they will handle, otherwise we should check with dartlang proj

https://github.com/TOPdesk/dart-junitreport/issues/15

**Comment by raubreysmith** [comment]
@archiebw-amido I've merged your latest changes from develop. Could you review please? Feel free to update the steps or naming.

**Review by archiebw-amido — APPROVED** [review]
All looks good to me.

**Pull request merged**